### PR TITLE
feat(appsync): L2 construct for EventBridge DataSource.

### LINKED ISSUE #24809: AWS AppSync: add EventBridge as L2 datasource
### Describe the feature

This feature adds eventBridge as an L2 construct following the release of the [L1 construct](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_appsync.CfnDataSource.EventBridgeConfigProperty.html)

### Use Case

Customers should have an easy way to add EventBridge as a datasource alongside the current L2 datasource options.

### Proposed Solution

```ts
// datasource.ts
/**
 * Properties for an AppSync EventBridge datasource
 */
export interface EventBridgeDataSourceProps extends BackedDataSourceProps {
	/**
	 * The EventBridge function to call to interact with this data source
	 */
	readonly eventBridgeBus: IEventBus
}

/**
 * An AppSync datasource backed by a EventBridge function
 */
export class EventBridgeDataSource extends BackedDataSource {
	constructor(scope: Construct, id: string, props: EventBridgeDataSourceProps) {
		super(scope, id, props, {
			type: 'AWS_EVENTBRIDGE',
			eventBridgeConfig: {
				eventBridgeBusArn: props.eventBridgeBus.eventBusArn,
			},
		})
		props.eventBridgeBus.grantInvoke(this)
	}
}
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

0.00.0

### Environment details (OS name and version, etc.)

mac m1 mbp

### PR DESCRIPTION
**Pull request by postsa**
Adds an L2 EventBridge data source to AppSync.

Closes #24809.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by andres-sinapsis** [comment]
Looks really good to me, hoping to have this soon!

**Comment by mtliendo** [comment]
Thanks so much @postsa for taking this on and having this close my issue/PR! Would love to give you a shoutout on Twitter/LinkedIn if you want to drop it here. Also next time I'm in Denver, 🍺 on me!

**Review by mtliendo — APPROVED** [review]

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 1]
```diff
@@ -232,6 +232,83 @@ httpDs.createResolver('MutationCallStepFunctionResolver', {
 });
 ```
 
+### EventBridge
+
+GraphQL schema file `schema.graphql`:
```
Comment (MrArnoldPalmer) [inline packages/aws-cdk-lib/aws-appsync/README.md:237]: Not being familiar with the event bridge integration, based on the mapping templates/schema, it looks like most of these types are just raw inputs/outputs from event bridge. Can we explain that in the docs a little bit?

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 2]
```diff
@@ -232,6 +232,83 @@ httpDs.createResolver('MutationCallStepFunctionResolver', {
 });
 ```
 
+### EventBridge
```
Comment (MrArnoldPalmer) [inline packages/aws-cdk-lib/aws-appsync/README.md:235]: Can we add 1 or 2 sentences saying when using event bridge + appsync is beneficial. A general use case etc?

We are probably missing these for other integrations but still a good thing to add.

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]
API is looking good to me. Some minor docs requests.

**Comment by postsa** [comment]
@MrArnoldPalmer I took a crack at those README updates. Thanks for the suggestions.

**Review by MrArnoldPalmer — APPROVED** [review]
:shipit:

**Pull request merged**